app/objects/secondclass/c_link.py

```diff
@@ -29,14 +29,19 @@ class Meta:
     unique = ma.fields.String()
     collect = ma.fields.DateTime(format='%Y-%m-%d %H:%M:%S', default='')
     finish = ma.fields.String()
-    # temp - replace with Nested(AbilitySchema)
-    ability = ma.fields.Function(lambda obj: obj.ability.display,
-                                 lambda obj: obj if isinstance(obj, Ability) else Ability.load(obj))
+    ability = ma.fields.Nested(AbilitySchema())
     cleanup = ma.fields.Integer(missing=0)
     visibility = ma.fields.Nested(VisibilitySchema)
     host = ma.fields.String(missing=None)
     output = ma.fields.String()
 
+    @ma.pre_load()
+    def fix_ability(self, link, **_):
+        if 'ability' in link and isinstance(link['ability'], Ability):
+            link_input = link.pop('ability')
+            link['ability'] = link_input.schema.dump(link_input)
```

Comment:
I can't use `link.pop('ability').display` here because Ability's display_schema excludes certain properties and will cause errors.